cores/arduino/pins_arduino_analog.h

```diff
@@ -0,0 +1,313 @@
+/*
+ *******************************************************************************
+ * Copyright (c) 2020, STMicroelectronics
+ * All rights reserved.
+ *
+ * This software component is licensed by ST under BSD 3-Clause license,
+ * the "License"; You may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at:
+ *                        opensource.org/licenses/BSD-3-Clause
+ *
+ *******************************************************************************
+ */
+#ifndef _PINS_ARDUINO_ANALOG_H_
+#define _PINS_ARDUINO_ANALOG_H_
+#include "variant.h"
+#include "PinNames.h"
+
+/* Arduino analog pins */
+#ifndef NUM_ANALOG_INPUTS
+  #define NUM_ANALOG_INPUTS 0
+#endif
+
+/*
+ * If NUM_ANALOG_FIRST is not defined:
+ * - Ax are not contiguous in the digitalPin array
+ * - analogInputPin array is available
+ */
+#ifndef NUM_ANALOG_FIRST
+  #define NUM_ANALOG_FIRST          (NUM_DIGITAL_PINS + 1)
+  #define NUM_ANALOG_LAST           (NUM_DIGITAL_PINS + NUM_ANALOG_INPUTS)
+  #define NUM_ANALOG_INTERNAL_FIRST (NUM_ANALOG_LAST + 1)
+#else
+  #define NUM_ANALOG_INTERNAL_FIRST (NUM_DIGITAL_PINS + 1)
+#endif
+
+/* ADC internal channels (not a pins) */
+/* Only used for analogRead() */
+#if defined(ADC_CHANNEL_TEMPSENSOR) || defined(ADC_CHANNEL_TEMPSENSOR_ADC1)
+  #define ATEMP        (NUM_ANALOG_INTERNAL_FIRST)
+#endif
+#ifdef ADC_CHANNEL_VREFINT
+  #define AVREF        (NUM_ANALOG_INTERNAL_FIRST + 2)
+#endif
+#ifdef ADC_CHANNEL_VBAT
+  #define AVBAT        (NUM_ANALOG_INTERNAL_FIRST + 3)
+#endif
+#if defined(ADC5) && defined(ADC_CHANNEL_TEMPSENSOR_ADC5)
+  #define ATEMP_ADC5   (NUM_ANALOG_INTERNAL_FIRST + 4)
+#endif
+
+/* If NUM_ANALOG_INPUTS is not defined there is no analog pins defined. */
+/* Anyway ADC internal channels are always available. */
+#if NUM_ANALOG_INPUTS > 0
+  #define PIN_A0       NUM_ANALOG_FIRST
+  static const uint8_t A0 = PIN_A0;
```

Comment:
3 years back it was an enum but it was requested for compatibility to move as a macro in order to be able to use preprocessor directive which is not possible with an enum: https://github.com/stm32duino/Arduino_Core_STM32/issues/140

Reply:
That issue is about `PIN_Ax`, which are now already macros and I was not suggesting to change those (because the names are longer, the chances of conflicts due to macros are a little smaller). My suggestion was to change the `Ax` constants to an enum. Not a very big deal, though.

Reply:
In fact we have mimic the way Arduino define Ax:
https://github.com/arduino/ArduinoCore-samd/blob/0b60a79c4b194ed2e76fead95caf1bbce8960049/variants/arduino_zero/variant.h#L94-L108
or 
https://github.com/arduino/ArduinoCore-mbed/blob/b86aaa794a3ca134ed727c710533ff253f3af539/variants/PORTENTA_H7_M4/pins_arduino.h#L26-L39

That's why I've aligned the Dx also on macro.
Anyway, I understood your point of view.

Reply:
> I assume a rebase is in order @matthijskooijman after my merging of #137?

Yeah, but I'd think it could be a (small) improvement for Arduino as well to switch from `const` to `enum` for the Ax constants.

> Anyway, I understood your point of view.

Ok :-)

Reply:
@matthijskooijman 
I've switched Ax to enum. :wink: